_posts/2022-01-28-filesystem-encryption-integrity.adoc

```diff
@@ -0,0 +1,145 @@
+---
+layout: post
+title: "Encrypting Filesystem realms and enabling integrity"
+date: 2022-01-28
+tags: filesystem-realm encryption integrity
+synopsis: An overview of the new filesystem realm encryption and integrity in the Elytron Security Realms.
+author: araskar
+link:
+---
+
+Previously, any data in the filesystem realm was always stored unencrypted and data was not verified. This meant that anyone with access to the filesystem identity files could access the credentials, and attributes (such as assigned roles). Along with that if anyone were to change the data in file, the realm would have no way of knowing. Now both encryption has been added via a SecretKey, and filesystem integrity has been added by generating a signature via a Public and Private Keypair.
+
+== An overview of the new attributes
+
+The filesystem realm now supports the following attributes
+
+* ``credential-store``: This attribute specifies the credential store resource where the secret key resides in. This credential store must be configured prior to creating the filesystem realm. This attribute is optional and if not specified, the filesystem realm will not be encrypted.
+
+* ``secret-key``: This attribute specifies the alias to the secret key in the credential store to encrypt the relam. By default the secret key alias is set to `key` when creating the credential store. This attribute is required if the `credential-store` attribute is specified.
```

Comment:
s/relam/realm

Reply:
s/By default the secret key alias is set to `key` when creating the credential store./The default value is `key`.

Reply:
Changed